tests/json/retryable_writes/bulkWrite.json

```diff
@@ -631,8 +631,8 @@
           "insertedIds": {
             "0": 2
           },
-          "matchedCount": 1,
-          "modifiedCount": 1,
+          "matchedCount": 0,
+          "modifiedCount": 0,
```

Comment:
Ah, I see why this is problematic. Since both update statements within the batch are adjacent, they're sent in the same `update` command. The original intention of this test wasn't to anticipate an error in the middle of a batch (unlike the insertMany tests), so I'd rather leave this as-is and re-order the bulk to: updateOne, insertOne, updateOne.